library/src/main/java/com/digi/xbee/api/connection/ConnectionType.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019, Digi International Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES 
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF 
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR 
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES 
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN 
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF 
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+package com.digi.xbee.api.connection;
+
+import java.util.HashMap;
+
+/**
+ * Enumerates the different connection types.
+ * 
+ * @since 1.3.0
+ */
+public enum ConnectionType {
+	
+	// Enumeration types.
+	SERIAL(0, "Serial connection"),
+	BLUETOOTH(1, "Bluetooth connection"),
+	UNKNOWN(3, "CoAP");
+
+	// Variables.
+	private int id;
+
+	private String name;
+
+	private static HashMap<Integer, ConnectionType> lookupTable = new HashMap<Integer, ConnectionType>();
+
+	static {
+		for (ConnectionType connectionType:values())
+			lookupTable.put(connectionType.getID(), connectionType);
+	}
+
+	/**
+	 * Class constructor. Instantiates a new {@code ConnectionType} enumeration
+	 * entry with the given parameters.
+	 *
+	 * @param id IP connection type ID.
+	 * @param name IP connection type name.
+	 */
+	private ConnectionType(int id, String name) {
+		this.id = id;
+		this.name = name;
+	}
+
+	/**
+	 * Retrieves the connection type ID.
+	 *
+	 * @return Connection type ID.
+	 */
+	public int getID() {
+		return id;
+	}
+
+	/**
+	 * Retrieves the connection type name.
+	 *
+	 * @return Connection type name.
+	 */
+	public String getName() {
+		return name;
+	}
+
+	/**
+	 * Retrieves the connection type for the given ID.
+	 *
+	 * @param id ID to retrieve the connection type.
+	 *
+	 * @return The connection type associated with the given ID.
+	 */
+	public static ConnectionType get(int id) {
+		return lookupTable.get(id);
```

Comment:
If we are not returning `UNKNOWN` if the given ID is not in the table, should we keep that entry?

Reply:
I'll change it to return `UNKNOWN` if the ID is not in the list.